.github/actions/trusted-checkout/checkout-submodules.sh

```diff
@@ -56,7 +126,8 @@ git submodule status | while read -r s; do
   fi
   echo "submodule at $path configured to use SSH - will reconfigure to https"
 
-  if ! $have_app; then
+  if [[ -z "${TOKEN_SERVER:-}" && ( -z "${APP_ID:-}" || -z "${APP_KEY:-}" ) ]]; then
+    echo "Neither a fed-server nor a GitHub App is specified, will try to fetch submodule anonymously"
```

Comment:
nit: might add explicit reference to input-name(s) to make error-message more obious